File: tests/test_analytics.rb
# I run a tracking pixel and analytics dashboard at analytics.alexwlchan.net
#
# This file has tests specific to this application.

require 'test/unit'

require_relative 'utils'

# Tests for analytics.alexwlchan.net
class TestAnalytics < Test::Unit::TestCase
  # The analytics app is running
  def test_homepage_is_up
    resp = get_url('https://analytics.alexwlchan.net/')

    assert_equal resp.code, '200'
  end

  # If you try to load the dashboard as a public user, you can't see it
  def test_dashboard_is_protected
    resp = get_url('https://analytics.alexwlchan.net/dashboard/')

    assert_equal resp.code, '302'
  end
end

File: uptime_tests/test_analytics.py
"""
Test my tracking pixel and analytics dashboard at analytics.alexwlchan.net.
"""

import httpx


def test_analytics_homepage() -> None:
    """
    The analytics app is running.
    """
    resp = httpx.get("https://analytics.alexwlchan.net/")

    assert resp.status_code == 200


def test_dashboard_is_protected() -> None:
    """
    If you try to load the dashboard as a public user, you can't see it.
    """
    resp = httpx.get("https://analytics.alexwlchan.net/dashboard/")

    assert resp.status_code == 302
    assert resp.headers["location"] == "/?next=%2Fdashboard%2F"
